packages/cli-kit/src/public/node/ruby.ts

```diff
@@ -443,15 +443,35 @@ async function getSpinEnvironmentVariables() {
  * @param directory - Directory where the Gemfile is located.
  */
 async function shopifyBundleInstall(directory: string): Promise<void> {
-  const bundle = bundleExecutable()
-  const shopifyGemsPath = shopifyGems.cache
-
-  await exec(bundle, ['config', 'set', '--local', 'path', shopifyGemsPath], {
+  await runBundler(['config', 'set', '--local', 'path', shopifyGems.cache], {
     cwd: directory,
   })
-
-  await exec(bundle, ['config', 'set', '--local', 'without', 'development:test'], {
+  await runBundler(['config', 'set', '--local', 'without', 'development:test'], {
     cwd: directory,
   })
-  await exec(bundle, ['install'], {cwd: directory})
+  await runBundler(['install'], {cwd: directory})
+}
+
+/**
+ * It returns a custom BUNDLE_USER_HOME. This is required in Windows because
+ * bundler will instead crash if the username contains UTF-8 characters.
+ *
+ * @returns The value of the environment variable.
+ */
+export function bundleUserHome(): string | undefined {
+  if (platformAndArch().platform === 'windows' && process.env.PUBLIC) {
```

Comment:
why do you need this env var to be present?

Reply:
because otherwise I'm not sure where the `Public` folder is, since it's windows it could be `C:\Users\Public` or `D:\Users\Public` etc :)